### PR TITLE
feat: message to merchant component

### PR DESCRIPTION
**Pull request by DilaraGueler**
## PR Type

[ ] Bugfix
[X ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no API changes)
[ ] Build-related changes
[ ] CI-related changes
[ ] Documentation content changes
[ ] Application / infrastructure changes
[ ] Other: 

## What Is the Current Behavior?

There is no message to merchant area during checkout.

## What Is the New Behavior?
Message to merchant component added on shipping page to the checkout process. Shoppers will be able to send messages to merchants.

## Does this PR Introduce a Breaking Change?

[ ] Yes
[X ] No

### Limitations
The user is not able to delete a message once he has already saved one. This is due to an issue in the REST API. The fix for it will be available with ICM 7.10.38.2 or higher.
We will enable the user to delete his message to merchant in the pwa after the ICM fix has been released..

[AB#81043](https://dev.azure.com/intershop-com/cefd1005-00a7-4c79-927f-a16947d1b2e6/_workitems/edit/81043)

### CODE REVIEW

**Review by SGrueber — CHANGES_REQUESTED** [review]
pls, add an e2e test

File: src/app/shared/components/basket/basket-merchant-message/basket-merchant-message.component.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  SimpleChange,
+  SimpleChanges,
+} from '@angular/core';
+import { FormGroup } from '@angular/forms';
+import { FormlyFieldConfig } from '@ngx-formly/core';
+
+import { CheckoutFacade } from 'ish-core/facades/checkout.facade';
+import { Basket } from 'ish-core/models/basket/basket.model';
+
+@Component({
```
Comment (SGrueber) [inline src/app/shared/components/basket/basket-merchant-message/basket-merchant-message.component.ts:17]: please add a short component documentation

**Review by SGrueber — CHANGES_REQUESTED** [review]

File: src/environments/environment.b2b.ts [hunk 2]
```diff
@@ -13,6 +13,7 @@ export const environment: Environment = {
     'businessCustomerRegistration',
     'costCenters',
     'maps',
+    'messageToMerchant',
```
Comment (SGrueber) [inline src/environments/environment.b2b.ts:16]: pls. mention the new feature toggle in the migrations.md

**Review by SGrueber — CHANGES_REQUESTED** [review]

**Review by marschmidt89 — APPROVED** [review]
no complaints regarding the .md file from my side

**Review by iwiederhold — CHANGES_REQUESTED** [review]
merchant_message_edit_missing

--------------------------------------------------------------------------------------------------------

merchant_message_label_spaces

**Review by schadii — APPROVED** [review]
Thanks!

**Review by SGrueber — APPROVED** [review]

**Review by schadii — DISMISSED** [review]

**Comment by iwiederhold** [comment]
After editing the merchant message again, you get an error.

image

**Comment by DilaraGueler** [comment]
>

**Review by SGrueber — APPROVED** [review]

**Pull request merged**